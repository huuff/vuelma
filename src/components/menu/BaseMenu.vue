<template>
<aside class="menu">
  <slot></slot>
</aside>
</template>

<script setup lang="ts">
import { toRef } from 'vue';
import { useOptionalTwoWayBinding } from '@/composables/optional-two-way-binding';
import { provideAccessors } from '@/composables/injected-accessors';
import partial from "lodash/partial";

const props = defineProps<{
  activeItem?: string;
}>();

const emit = defineEmits<{
  (event: "update:activeItem", id?: string): void
}>();

const actualActiveItem = useOptionalTwoWayBinding(undefined, toRef(props, 'activeItem'), partial(emit, "update:activeItem"));
provideAccessors("ActiveMenuItem", actualActiveItem);

</script>
