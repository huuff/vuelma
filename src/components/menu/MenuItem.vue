<template>
<slot></slot>
</template>

<script setup lang="ts">
import { OptionalId } from '@/util/optional-id';

export interface MenuItemProps extends OptionalId {
  titleText: string;
  itemId?: string;
}

const props = defineProps<MenuItemProps>();
</script>
