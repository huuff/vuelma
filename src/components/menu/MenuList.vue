<template>
<slot></slot>
</template>

<script setup lang="ts">

export type MenuListProps = {
  label: string;
};

const props = defineProps<MenuListProps>();
</script>
