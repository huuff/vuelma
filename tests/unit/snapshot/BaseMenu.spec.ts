import { mount } from "@vue/test-utils";
import BaseMenu from "@/components/menu/BaseMenu.vue";
import MenuList from "@/components/menu/MenuList.vue";
import MenuItem from "@/components/menu/MenuItem.vue";

test("BaseMenu.vue snapshot", () => {
  const wrapper = mount(BaseMenu, {
    props: {
      activeItemId: "Item 1",
    },
    global: {
      components: { MenuList, MenuItem },
    },
    slots: {
      default: `
        <menu-list label="List">
          <menu-item titleText="Item 1" />
          <menu-item titleText="Item 2" />
          <menu-list label="Stacked">
            <menu-item titleText="Item 3" />
            <menu-item titleText="Item 4" />
            <menu-item titleText="Item 5" />
          </menu-list>
        </menu-list>
      `,
    },
  });

  expect(wrapper.element).toMatchSnapshot();
});
