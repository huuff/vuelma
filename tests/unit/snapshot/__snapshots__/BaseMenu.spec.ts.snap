// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseMenu.vue snapshot 1`] = `
<aside
  class="menu"
>
  
  
  <p
    class="menu-label"
  >
    List
  </p>
  <ul
    class="menu-list"
  >
    <li>
      <a
        class="is-active"
        titletext="Item 1"
      >
        Item 1
      </a>
    </li>
    <li>
      <a
        class=""
        titletext="Item 2"
      >
        Item 2
      </a>
    </li>
    <li>
      <p
        class="menu-label"
      >
        Stacked
      </p>
      <ul
        class="menu-list"
      >
        <li>
          <a
            class=""
            titletext="Item 3"
          >
            Item 3
          </a>
        </li>
        <li>
          <a
            class=""
            titletext="Item 4"
          >
            Item 4
          </a>
        </li>
        <li>
          <a
            class=""
            titletext="Item 5"
          >
            Item 5
          </a>
        </li>
      </ul>
    </li>
  </ul>
  
  
</aside>
`;
